### ISSUE TITLE
Improve code coverage for Microsoft.VisualBasic.dll

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#14344]
There are [almost zero tests](https://github.com/dotnet/corefx/blob/master/src/Microsoft.VisualBasic/tests/StringsTests.cs) so there is [essentially zero code coverage](https://ci.dot.net/job/dotnet_corefx/job/master/job/code_coverage_windows/Code_Coverage_Report/).

**Comment by stephentoub** [comment]
> There are almost zero tests

The only reason there aren't exactly zero is because if there were zero, it wouldn't show up in the code coverage report at all, so I added one test to help highlight how bad we are here ;)

**Comment by dennisdietrich** [comment]
@stephentoub Well, that's pretty bad. ;) I could take this one. Is there an existing test codebase somewhere I could use as a starting point or do I have to start from scratch (I mean, from the one test you added :))?

**Comment by stephentoub** [comment]
Thanks, @dennisdietrich.  That's a nice offer.  @VSadov, do you know if there are existing tests for this internally somewhere?

**Comment by danmosemsft** [comment]
Maybe from whatever repo @JeremyKuhne found the serial port tests recently.

**Comment by VSadov** [comment]
I will take a look if there is something salvageable.

**Comment by danmosemsft** [comment]
@VSadov any luck?

**Comment by danmosemsft** [comment]
Info on code coverage work:

1) Here's the all up report
https://ci.dot.net/job/dotnet_corefx/job/master/job/code_coverage_windows/Code_Coverage_Report/
Caveat - doesn't show coverage of any types whose implementation is in corelib (eg String, etc). If you want to look at that for now you must gather manually using instructions below.

2) Docs on code coverage work are here
https://github.com/dotnet/corefx/blob/master/Documentation/building/code-coverage.md
and some in
https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/developer-guide.md

3) There's innumerable places to add coverage -- we would love to focus on the most heavily used types and the least covered types to get the most benefit.

**Comment by ehasis** [comment]
I'm planing to contribute with this issue, but for some reason the there is no coverage (0 classes) for this assembly (actually Microsoft.VisualBasic.Core.dll). Any help in how to make code coverage show for this assembly?

**Comment by danmosemsft** [comment]
@ehasis did you try https://github.com/dotnet/corefx/blob/master/Documentation/building/code-coverage.md#local-code-coverage-runs

**Comment by ehasis** [comment]
@danmosemsft yes, I tried that. I also tried test with coverage on the entire repository, the report showed for all the other assemblies, except the VisualBasic one.

**Comment by danmosemsft** [comment]
@viktorhofer is VB supposed to work? I would expect the tools to not be aware of the language used.

**Comment by ViktorHofer** [comment]
Seems to be a coverlet bug:

```
Unable to instrument module: C:\git\corefx2\artifacts\bin\testhost\netcoreapp-Windows_NT-Debug-x64\shared\Microsoft.NETCore.App\5.0.0\Microsoft.VisualBasic.Core.dll because : Object reference not set to an instance of an object.
```

cc @MarcoRossignoli

**Comment by MarcoRossignoli** [comment]
I'm OOF for a week I'll take a look asap @ViktorHofer please can you attach command line/project folder to speed up repro?

**Comment by ViktorHofer** [comment]
Submitted https://github.com/tonerdo/coverlet/pull/495 to fix the crash.

**Comment by ViktorHofer** [comment]
@ehasis that's the current code coverage status for Microsoft.VisualBasic.Core:

```
+----------------------------+--------+--------+--------+
| Module                     | Line   | Branch | Method |
+----------------------------+--------+--------+--------+
| Microsoft.VisualBasic.Core | 55.04% | 54.24% | 62.75% |
+----------------------------+--------+--------+--------+

+---------+--------+--------+--------+
|         | Line   | Branch | Method |
+---------+--------+--------+--------+
| Total   | 55.04% | 54.24% | 62.75% |
+---------+--------+--------+--------+
| Average | 55.04% | 54.24% | 62.75% |
+---------+--------+--------+--------+
```

HTML report:
[report.zip](https://github.com/dotnet/corefx/files/3419279/report.zip)

**Comment by ehasis** [comment]
Very nice, thanks @ViktorHofer !

**Comment by MarcoRossignoli** [comment]
Merged thanks Viktor

**Comment by ViktorHofer** [comment]
Submitted https://github.com/dotnet/corefx/pull/39728 to consume the new version.
